lib/ansible/modules/source_control/git.py

```diff
@@ -1073,7 +1108,7 @@ def main():
                     result['diff'] = diff
             module.exit_json(**result)
         # there's no git config, so clone
-        clone(git_path, module, repo, dest, remote, depth, version, bare, reference, refspec, verify_commit)
```

Comment:
You should also extend check mode handling for `separate_git_dir` arg.

Reply:
I think the check code for the clone already handles everything, it just returns 
```{
     change: True, 
     after: HEAD_OF_REMOTE
    }```
so it doesn't matter whether I add a new argument to clone. What's your opinion?

Reply:
I think we could add a few additional return fields to indicate the location of the bare repo.